### PR TITLE
Chrome Extension Enumeration

### PR DESCRIPTION
**Pull request by valrkey**
Plugin requested by @mikeclemson.

@5ynax: This plugin is an edge case for what we have written so far so I'm not exactly sure how to fit this to the plugin template we have going so far. It doesn't copy files back so normally I'd call this a Collect plugin, but it requires post-processing of sending a bunch of web requests to resolve the Chrome extension IDs with Google. Right now I have it as a Retrieve plugin to allow for the web requests to be generated on the analyst machine rather than the machine under investigation.

### CODE REVIEW

File: Plugins/Persistence/Retrieve-ChromeExtensions.ps1 [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    # Get all the chome extensions
+    $ChromeExtensions = @(
+        'C:\Users\*\AppData\Local\Google\Chrome\User Data\Default\Extensions\*',
+        'C:\Users\*\AppData\Local\Chromium\User Data\Default\Extensions\*'
+    )
+
+    # Gather the extension list
+    $Extensions = Invoke-Command -Session $Session -ScriptBlock { Get-Item -Path $using:ChromeExtensions }
+
+    # $Extensions | Foreach-Object { Add-Member -InputObject $PSItem -NotePropertyName 'ExtensionName' -NotePropertyValue () }
```
Comment (5ynax) [inline Plugins/Persistence/Retrieve-ChromeExtensions.ps1]: Is this needed?

Comment (5ynax) [reply Plugins/Persistence/Retrieve-ChromeExtensions.ps1]: The commented out line starting at 68

File: Plugins/Persistence/Retrieve-ChromeExtensions.ps1 [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    return $Extensions | Select-Object -Property @{Name='ExtensionID'; Expression={$PSItem.Name}},@{Name='ExtensionName'; Expression={Resolve-ChromeExtension -ExtensionID $PSItem.Name}},@{Name='User'; Expression={$PSItem.FullName -Split '\\' | Select-Object -First 1 -Skip 2}},@{Name='Path'; Expression={$PSItem.FullName}},'LastWriteTime','PSComputerName'
```
Comment (5ynax) [inline Plugins/Persistence/Retrieve-ChromeExtensions.ps1]: so ffaaaaaaaancy

File: Plugins/Persistence/Retrieve-ChromeExtensions.ps1 (hunk 1 above)
Comment (valrkey) [reply Plugins/Persistence/Retrieve-ChromeExtensions.ps1]: Lol probably not

File: Plugins/Persistence/Retrieve-ChromeExtensions.ps1 (hunk 2 above)
Comment (valrkey) [reply Plugins/Persistence/Retrieve-ChromeExtensions.ps1]: Much fance! So wow!

File: Plugins/Persistence/Retrieve-ChromeExtensions.ps1 (hunk 1 above)
Comment (valrkey) [reply Plugins/Persistence/Retrieve-ChromeExtensions.ps1]: I fixed da thing

**Pull request merged**